test/DurableTaskLifeCycleNotificationTest.cs

```diff
@@ -0,0 +1,361 @@
+                    mock.Protected()
+                        .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                        .Returns((HttpRequestMessage request, CancellationToken cancellationToken) =>
+                        {
+                            Assert.True(request.Headers.Any(x => x.Key == "aeg-sas-key"));
+                            var values = request.Headers.GetValues("aeg-sas-key").ToList();
+                            Assert.Single(values);
+                            Assert.Equal(eventGridKey, values[0]);
+                            Assert.Equal(eventGridEndpoint, request.RequestUri.ToString());
+                            var json = request.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+                            dynamic content = JsonConvert.DeserializeObject(json);
+                            foreach (dynamic o in content)
+                            {
+                                Assert.Equal("1.0", o.dataVersion.ToString());
+                                Assert.Equal(nameof(this.OrchestrationStartAndCompleted), o.data.HubName.ToString());
+                                Assert.Equal(orchestratorFunctionNames[0], o.data.FunctionName.ToString());
+
+                                if (callCount == 0)
+                                {
+                                    Assert.Equal("durable/orchestrator/Running", o.subject.ToString());
+                                    Assert.Equal("orchestratorEvent", o.eventType.ToString());
+                                    Assert.Equal("0", o.data.EventType.ToString());
+                                }
+                                else if (callCount == 1)
+                                {
+                                    Assert.Equal("durable/orchestrator/Completed", o.subject.ToString());
+                                    Assert.Equal("orchestratorEvent", o.eventType.ToString());
+                                    Assert.Equal("1", o.data.EventType.ToString());
+                                }
+                                else
+                                {
+                                    throw new Exception("Call Count is Bad");
+                                }
+                            }
+
+                            callCount++;
+                            return Task.FromResult(new HttpResponseMessage(HttpStatusCode.InternalServerError));
+                        });
+
+                    extension.LifeCycleTraceHelper.SetHttpMessageHandler(mock.Object);
+                }
+
+                var client = await host.StartOrchestratorAsync(orchestratorFunctionNames[0], "World", this.output);
+                var status = await client.WaitForCompletionAsync(TimeSpan.FromSeconds(30), this.output);
+
+                Assert.Equal(OrchestrationRuntimeStatus.Completed, status?.RuntimeStatus);
```

Comment:
Just to make sure I understand: the intent of this test is to ensure that the orchestration completes even if the Event Grid notification fails, correct? Should we also check the trace output to make sure the new trace message is written? Several of the tests in DurableTaskEndToEndTests.cs do similar logging validation.

Reply:
Yes, you are right. We added the testing of the trace messages. Thank you!